protocols/imap/src/main/java/org/apache/james/imap/main/PathConverter.java

```diff
@@ -47,49 +48,48 @@ private PathConverter(ImapSession session) {
     }
 
     public MailboxPath buildFullPath(String mailboxName) {
```

Comment:
As usual I don't have any clue what is supposed to come in this string. As the magic happen on special values of this string, I'm worried people get confused and pass anything to this method and have bad results

Reply:
Well, we should locate the parsing of mailbox name somewhere, no?

Futhermore this design is the one presented on grooming, ticket description and sprint planning.

Reply:
You want javadoc?

Reply:
The fact that I need to read the code for an hour or so to fully understand things is the sign names are not so good.
I suggest a small design change : 
1. create a MailboxName class (or whatever the name is) that we will use to strongly type the incoming mailboxName input in IMAP Layer. 
2. Have a `toMailboxPath(Session)` method on this MailboxName object (we would take the code from current PathConverter that)
3. Have a factory method `MailboxName.from(MailboxPath)` to go back to MailboxName before displaying them to the user.

I think that we gain type security by explicitely wrapping the mailboxName String into a MailboxName object before going ahead. Then we also prevent users from building a MailboxPath using PathConverter without understanding what the required String format is.

WDYT ?

Reply:
Well you lost some pathDelimiter and namespace configuration along the way. ( That might make this nice api proposal a bit more ugly )


I agree with such strong typing that complete the aporoach that @aduprat and I have followed for the past week. Location of that code is imo not that important.